server/aesdsocket.c

```diff
@@ -0,0 +1,379 @@
+
+        if (client_fd == -1) {
+            // Ignore bad file descriptor error when shutdown starts
+            if (errno != EBADF) {
+                close(socket_fd);
+                syslog(LOG_ERR, "Error accept(): %s\n", strerror(errno));
+            }
+            exit_status = 1;
+            continue;
+        }
+
+        // Log message to syslog "Accepted connection from <CLIENT_IP_ADDRESS>"
+        memset(client_ip, 0, sizeof(client_ip));
+        inet_ntop(client_addr.ss_family, 
+                    get_in_addr((struct sockaddr*)&client_addr), 
+                    client_ip, 
+                    sizeof(client_ip));
+        syslog(LOG_INFO, "Accepted connection from %s\n", client_ip);
+
+        int rx_size = READ_SIZE;
+        int rx_bytes = 0;
+        char* rx_buffer = malloc(rx_size);
+
+        if (rx_buffer == NULL) {
+            syslog(LOG_ERR, "Error failed to malloc()\n");
+            close(socket_fd);
+            close(client_fd);
+            exit_status = 1;
+            continue;
+        }
+
+        // Zero memory before reading
+        memset(rx_buffer, 0, rx_size);
+        int total_bytes = 0;
+        int bytes_to_read = 0;
+        char* p_end = NULL;
+
+        // Create file to write packets to
+        FILE* data_file = fopen(TMP_FILE, "a+");
+
+        if (data_file == NULL) {
+            syslog(LOG_ERR, "Error fopen(): %s\n", strerror(errno));
+            close(socket_fd);
+            close(client_fd);
+            if (rx_buffer != NULL) {
+                free(rx_buffer);
+                rx_buffer = NULL;
+            }
+            exit_status = 1;
+            continue;
+        }
+
+        while (1) {
+
+            bytes_to_read = rx_size - total_bytes;
+
+            if (bytes_to_read == 0) {
+                // Increase size of rx_buffer
+                rx_size += READ_SIZE;
+                bytes_to_read = READ_SIZE;
+                rx_buffer = realloc(rx_buffer, rx_size);
+                if (rx_buffer == NULL) {
+                    syslog(LOG_ERR, "Error failed to realloc()\n");
+                    fclose(data_file);
+                    close(socket_fd);
+                    close(client_fd);
+                    break; // send to cleanup
+                }
+                memset(&(rx_buffer[total_bytes]), 0, READ_SIZE);
+            }
+
+            rx_bytes = recv(client_fd, &(rx_buffer[total_bytes]), rx_size - total_bytes, 0);
+
+            if (rx_bytes == 0) {
+                break; // done sending
+            }
+
+            total_bytes += rx_bytes;
+
+            // check buffer for '\n' newline character
+            // If it exists, append to file and send back on client 
+            // check for all '\n' characters, after transmitting
+            // move all data down to zero index of malloc'd packet
+            // USE strchr() to fine \n characters: 
+            // https://man7.org/linux/man-pages/man3/strchr.3.html
+            if (total_bytes < rx_size) {
```

Comment:
@mskogen what happens when total_bytes are >= rx_size?  In other words, what happens if more than 1024 bytes are sent before `\n`?